contentcuration/automation/models.py

```diff
@@ -1,3 +1,30 @@
-# from django.db import models
+import uuid
 
-# Create your models here.
+from django.db import models
+from kolibri_public.models import ContentNode
+
+
+REQUEST_HASH_INDEX_NAME = "request_hash_idx"
+CONTENTNODE_ID_INDEX_NAME = "contentnode_id_idx"
+
+
+class RecommendationsCache(models.Model):
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    request_hash = models.CharField(max_length=32, null=True)
+    contentnode_id = models.ForeignKey(
+        ContentNode,
+        null=True,
+        blank=True,
+        related_name='recommendations',
+        on_delete=models.SET_NULL,
```

Comment:
What's the reasoning for setting this to null on deletion of the node?

Reply:
In light of your comment [here](https://github.com/learningequality/studio/pull/4604#discussion_r1700298183) i guess it defeats to purpose setting it null and keeping the record. I think a better approach would be to cascade the deletion?

Reply:
Yeah I think cascade deletion should be fine.